.github/workflows/ci.yml

```diff
@@ -108,12 +124,14 @@ jobs:
       - cargo
       - cargo-toml-fmt-check
       - cargo-unused-deps-check
+      - publish-always-check
     runs-on: ubuntu-latest
     steps:
       - run: true
 
-  publish-check:
+  publish-final-check:
```

Comment:
We should split this job into two parts, tag verification vs actual crate publishing. This name seems misleading as we are doing more than checking things here, this job will have side-effects by doing actual publishing